stout/range.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <algorithm>
+
+#include "stout/stream.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+struct _Range {
+  template <typename K_, typename Arg_>
+  struct Continuation : public detail::TypeErasedStream {
+    // NOTE: explicit constructor because inheriting 'TypeErasedStream'.
+    Continuation(K_ k, int from, int to, int step)
+      : k_(std::move(k)),
+        to_(to),
+        from_(std::min(from, to)),
+        step_(step) {}
```

Comment:
Small thing: let's keep the order of `from`, `to` and `step` consistent - same order in the constructor, parameter lists, and definition.